src/ImageSharp.Web/Providers/PhysicalFileSystemProvider.cs

```diff
@@ -70,8 +70,7 @@ public Task<IImageResolver> GetAsync(HttpContext context)
                 var fileInfo = new FileInfo(fullPath);
                 if (fileInfo.Exists)
                 {
-                    var metadata = new ImageMetadata(fileInfo.LastWriteTimeUtc, fileInfo.Length);
```

Comment:
It's actually ok to create the metadata here. `IImageProvider.GetAsync(HttpContext context)` is called exactly one time and the type is only initialized if the source file exists. By doing it like this we avoid the cost of the struct creation per per request since it's a readonly struct.